extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/graal/Brotli4jFeature.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.vertx.http.runtime.graal;
+
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeClassInitialization;
+import org.graalvm.nativeimage.hosted.RuntimeResourceAccess;
+
+import io.quarkus.runtime.graal.GraalVM;
+import io.quarkus.utilities.OS;
+
+public class Brotli4jFeature implements Feature {
+
+    @Override
+    public void beforeAnalysis(BeforeAnalysisAccess access) {
+        // TODO reuse Brotli4jLoader's code instead of reinventing the wheel?
```

Comment:
I think this makes a lot of sense. In most cases we were reinventing the wheel, we were doing it because we couldn't actually resolve the OS at build time for in container build.

By using this pattern, we can probably do it now.

Now I can understand if you want to do it in two steps.